### PR TITLE
Add gettext-based translations

### LINKED ISSUE #97: Add translation plugin
Should we add translated string in this lib or make the devs add their own?
I like the idea of a standard coming directly from here.
It would probably be stuff like "No more results" or "An error has occurred" I guess. No big sentences, but still.

We should make sure developpers can dynamically import stuff from us so they don't load every language then :)

Any opinions?
@nextcloud/vue

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes #97 

Todo

- [x] Test with a real app
- [x] Test placeholder replacement (not sure if `%s` or `{placeholder}` should be used
- [ ] Set up https://docs.transifex.com/transifex-github-integrations/github-tx-ui

### CODE REVIEW

File: build/extract-l10n.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            arguments: {
+                text: 0,
+                context: 1
+            }
+        }),
+        JsExtractors.callExpression('n', {
+            arguments: {
+                text: 1,
+                textPlural: 2,
+                context: 3
```
Comment (nickvergessen) [inline build/extract-l10n.js]: Actually count js 3 and context is 4

Comment (ChristophWurst) [reply build/extract-l10n.js]: Yep, count is definitely 3. I think we don't even use context. At least not for `ngettext` https://github.com/alexanderwallin/node-gettext#Gettext

**Comment by ChristophWurst** [comment]
> Test placeholder replacement (not sure if `%s` or `{placeholder}` should be used

We'll eventually need this for every package that is translated, so I added this functionality to the l10n package: https://github.com/nextcloud/nextcloud-l10n/pull/88

**Comment by nickvergessen** [comment]
> Test placeholder replacement (not sure if %s or {placeholder} should be used

JS uses `{}` + `%n`
PHP uses `%s`, `%1$s` + `%n`

Ref https://github.com/nextcloud/documentation/pull/1827

**Comment by ChristophWurst** [comment]
> JS uses `{}` + `%n`

Done via https://github.com/nextcloud/nextcloud-l10n/pull/88. Please have a look :)

**Pull request merged**